src/query/anonymization.c

```diff
@@ -150,6 +151,197 @@ static void rewrite_to_anon_aggregator(Aggref *aggref, List *aid_refs, Oid fnoid
   append_aid_args(aggref, aid_refs);
 }
 
+/* Intended for the denominator in the `avg(col)` rewritten to `sum(col) / count(col)`. */
+static Aggref *make_anon_count_value_aggref(const Aggref *source_aggref)
+{
+  Aggref *count_aggref = makeNode(Aggref);
+  count_aggref->aggfnoid = g_oid_cache.anon_count_value;
+  count_aggref->aggtype = INT8OID;
+  count_aggref->aggcollid = source_aggref->aggcollid;
+  count_aggref->inputcollid = source_aggref->inputcollid;
+  count_aggref->aggtranstype = InvalidOid; /* Will be set by planner. */
+  count_aggref->aggargtypes = list_copy(source_aggref->aggargtypes);
+  count_aggref->aggdirectargs = list_copy(source_aggref->aggdirectargs);
+  count_aggref->args = list_copy(source_aggref->args);
+  count_aggref->aggorder = list_copy(source_aggref->aggorder);
+  count_aggref->aggdistinct = source_aggref->aggdistinct;
+  count_aggref->aggfilter = source_aggref->aggfilter;
+  count_aggref->aggstar = false;
+  count_aggref->aggvariadic = source_aggref->aggvariadic;
+  count_aggref->aggkind = source_aggref->aggkind;
+  count_aggref->agglevelsup = source_aggref->agglevelsup;
+  count_aggref->aggsplit = source_aggref->aggsplit;
+  count_aggref->location = source_aggref->location;
```

Comment:
This seems a bit brittle. The tree mutator handles copying, or we can use `copyObject` and only change what we need.

Reply:
agreed, I was looking at `copyObject`, but I wasn't sure if it would work in this case.